docs/IsolationNetworkPolicies.md

```diff
@@ -0,0 +1,210 @@
+- Currently, Fybrik assumes co-location of client workloads with data plane entry point modules. Future implementations
+might support deployment workloads on different clusters or run them out of Kubernetes as a standalone applications.
+  - We check the co-location of a Fybrik module and a relevant workload in the same cluster as a specific use case.
+- We should allow workload client connections isolation based on IP addresses. It can be useful if a workload is not 
+co-locate with a Fybrik data plane.  
+
+## Network Policies
+
+Fybrik is a Kubernetes application, therefore, the simplest isolation method can be based on 
+[Kubernetes Network Policies](https://kubernetes.io/docs/concepts/services-networking/network-policies/) (NP).
+NP allow to control traffic flow at the IP address or port level (OSI layer 3 or 4). NP can restrict incoming (ingress) 
+and/or outgoing (egress) communications. 
+The restrictions are based on combinations of Pod labels, Namespaces labels and IP blocks.
+
+### Advantages:
+- does not require any changes in modules implementations, can be implemented in the Fybrik control plane only.
+- can be combined with other isolation methods
+
+### Disadvantages:
+- a single Kubernetes cluster oriented, cross clusters restrictions can be based on IP blocks only, which significantly 
+complicates the configuration. Therefore, later we will provide other isolation methods as an extension to NP.
+- NP are implemented by the network plugin. Creating a NetworkPolicy resource without a controller - a Container Network 
+Interface (CNI) that implements it will have no effect. Kind deployments which by default use "kindnetd" do not
+support NP. Support NP requires installation another CNI, e.g. [Calico](https://github.com/projectcalico/calico). Here is the
+[instructions](https://alexbrand.dev/post/creating-a-kind-cluster-with-calico-networking/). On the other hand, all K8s 
+clusters in production deployments use CNIs which support NP.
+
+## Implementation
+
+The `from` element of `ingress` 
+[NetworkPolicySpec](https://kubernetes.io/docs/reference/kubernetes-api/policy-resources/network-policy-v1/#NetworkPolicySpec) 
+allows to restrict incoming network traffic based on `podSelectors` (labels based), and/or 
+`namespaceSelectors` (labels based) and/or `ipBlocks`. The last one permits incoming network traffic from predefined IP 
+address blocks. 
+The `to` element of `egress` restricts outgoing connections and has similar structure. 
+
+Current `FybrikApplicationSpec` has the “[selector](https://fybrik.io/v1.2/reference/crds/#fybrikapplicationspecselector)” 
+element, which is a combination of `clusterName` and `workloadSelector`. Unfortunately, based on this information we can 
+implement only one label-based `podSelector`, which will allow incoming connections from pods in the namespace where the NP 
+instance is deployed and have the defined label. In order to restrict incoming connections to Fybrik modules, the NP
```

Comment:
There is no mention of who deploys the NP nor how it is decided where to deploy them, so this problem is unclear to me.

Reply:
See Line: 184 "Extend blueprint’s module install/uninstall process by installation/removing of relevant NP instances."

This problem is a consequent of the NP [spec](https://kubernetes.io/docs/concepts/services-networking/network-policies/), you are welcome to take a look at it, plus it was presented on one of our group meetings.